src/stores/events/EventsDataStore.ts

```diff
@@ -320,7 +331,11 @@ export default class EventsDataStore {
 
 				const { isUnknown, parentId, eventId } = event;
 
-				if ((isRootEvent(event) || isUnknown) && !rootNodes.includes(eventId)) {
+				if (
+					(isRootEvent(event) || isUnknown) &&
```

Comment:
now it gets a little bit simplier we want to add to rootEventIds only if its isRootEvent and !rootNodes.includes(rootNode.eventId)